### ISSUE TITLE
union(): Union wrong behavior when type validation pass

### ISSUE THREAD

**Issue by alexabw** [fabian-hiller/valibot#364]
# union(): Union wrong behavior when type validation pass

If we have a union with nested validation for a number (just for example), we will get an `Invalid type` error for union, which is incorrect, because the type is correct, only the specific value of this type is incorrect (so `actionIssue` returned and `result.typed === true`).
Next, validation for the literal will start, although the union has already obtained a result where `typed === true`

```ts
const schema= union(
  [number([maxValue(5, 'Max 5')]), literal('')]
);

safeParse(schema, 10);
```

```json
{
    "typed": false,
    "success": false,
    "data": 10,
    "output": 10,
    "error": {
        "issues": [
            {
                "reason": "type",
                "validation": "union",
                "origin": "value",
                "message": "Invalid type",
                "input": 10,
                "issues": [
                    {
                        "reason": "number",
                        "validation": "max_value",
                        "origin": "value",
                        "message": "Max 5",
                        "input": 10,
                        "requirement": 5
                    },
                    {
                        "reason": "type",
                        "validation": "literal",
                        "origin": "value",
                        "message": "Invalid type",
                        "input": 10
                    }
                ]
            }
        ],
        "name": "ValiError"
    },
    "issues": [
        {
            "reason": "type",
            "validation": "union",
            "origin": "value",
            "message": "Invalid type",
            "input": 10,
            "issues": [
                {
                    "reason": "number",
                    "validation": "max_value",
                    "origin": "value",
                    "message": "Max 5",
                    "input": 10,
                    "requirement": 5
                },
                {
                    "reason": "type",
                    "validation": "literal",
                    "origin": "value",
                    "message": "Invalid type",
                    "input": 10
                }
            ]
        }
    ]
}
```

It also breaks pipeline execution of union, so custom is not executed
```ts
const schema= union(
  [number([maxValue(5, 'Max 5')]), literal('')],
  [custom((value) => value !== 10, 'Error from pipeline')]
);

safeParse(schema, 10);
```

# Expected behavior

Return errors only for the first schema that pass type check
Or if all failed with type error, then return all errors

Same behavior we have in zod union's

# Possible fix

Add check for `result.typed` in union's `_parse()` function

```ts
_parse(input, info) {
  // Create issues and output
  let issues: Issues | undefined;
  let output: [Output<TOptions[number]>] | undefined;

  // Parse schema of each option
  for (const schema of this.options) {
    const result = schema._parse(input, info);

    // Check for result.typed here ---------------------------------------------------------------
    if (result.typed) {
      issues = result.issues;
      output = [result.output];
      break;
    }
    // ----------------------------------------------------------------------------------------------

    // If there are issues, capture them
    if (result.issues) {
      if (issues) {
        for (const issue of result.issues) {
          issues.push(issue);
        }
      } else {
        issues = result.issues;
      }

      // Otherwise, set output and break loop
    } else {
      // Note: Output is nested in array, so that also a falsy value
      // further down can be recognized as valid value
      output = [result.output];
      break;
    }
  }

  // If there is an output, execute pipe
  if (output) {
    return pipeResult(output[0], this.pipe, info, 'union', issues);
  }

  // Otherwise, return schema issue
  return schemaIssue(
    info,
    'type',
    'union',
    this.message,
    input,
    undefined,
    issues
  );
},
```

**Comment by fabian-hiller** [comment]
Thank you for reporting this issue! It is right that the current implementation can potentially break the pipeline execution. I will investigate this.

> Return errors only for the first schema that pass type check

This is not possible because it would break the functionality of `union` for any schema where the same schema function occurs twice as part of the union. Consider the following schema.

```ts
import * as v from 'valibot';

const Schema = v.union([
  v.number([v.minValue(2), v.maxValue(4)]),
  v.number([v.multipleOf(10)]),
]);

const result = v.safeParse(Schema, 20);
```

`20` is a multiple of `10` but if we break the loop after the first schema function, the second will never be executed.

**Comment by fabian-hiller** [comment]
Thank you for your contribution! This should be fixed in v0.26.0.
